objects.js

```diff
@@ -2634,96 +2634,149 @@ SpriteMorph.prototype.searchBlocks = function (
     if (searchString) {searchPane.reactToKeystroke(); }
 };
 
-// SpritMorph parsing simple arithmetic expressions to reporter blocks
+// SpriteMorph parsing simple arithmetic expressions to reporter blocks
 
 SpriteMorph.prototype.reporterize = function (expressionString) {
     // highly experimental Christmas Easter Egg 2016 :-)
     var ast;
 
-    function parseInfix(expression, operator, already) {
-        // very basic diadic infix parser for arithmetic expressions
-        // with strict left-to-right operator precedence (like in Smalltalk)
-        // which can be overriden by - nested - parentheses.
-        // assumes well-formed expressions, no graceful error handling yet.
-
-        var inputs = ['', ''],
-            idx = 0,
-            ch;
-
-        function format(value) {
-            return value instanceof Array || isNaN(+value) ? value : +value;
-        }
-
-        function nested() {
-            var level = 1,
-                expr = '';
-            while (idx < expression.length) {
-                ch = expression[idx];
-                idx += 1;
-                switch (ch) {
-                case '(':
-                    level += 1;
-                    break;
-                case ')':
-                    level -= 1;
-                    if (!level) {
-                        return expr;
-                    }
-                    break;
-                }
-                expr += ch;
+    function tokenize(expressionString) {
+        // very basic tokenizer for arithmetic expressions
+        var tokens = [];
+        var DIGITS = /\d|\./;
+        var IDENT_START = /[A-Za-z_]/;
```

Comment:
Is this being used to match variable names? If so it seems like we should greatly expand what it accepts, though that does significant complicate things if it were to match spaces and operators, but perhaps we can ignore that for now...